presto-main/src/main/java/com/facebook/presto/execution/QueryStateMachine.java

```diff
@@ -752,6 +753,23 @@ public void setUpdateType(String updateType)
         this.updateType.set(updateType);
     }
 
+    public long getScannedBytes(StageInfo rootStage)
+    {
+        long rawInputDataSize = 0;
+        for (StageInfo stageInfo : getAllStages(Optional.of(rootStage))) {
+            if (stageInfo.getPlan().isPresent()) {
+                PlanFragment plan = stageInfo.getPlan().get();
+                if (!plan.getTableScanSchedulingOrder().isEmpty()) {
+                    Optional<QueryInfo> queryInfo = finalQueryInfo.get();
+                    if (queryInfo.isPresent()) {
+                        QueryStats queryStats = queryInfo.get().getQueryStats();
+                        rawInputDataSize += queryStats.getRawInputDataSize().toBytes();
+                    }
+                }
+            }
+        }
```

Comment:
I can see a lot of issues with this logic:

1. We are reading `finalQueryInfo` which is published only when the query is finished - can you test and see if this number is actually published when the query is running ?
2. For every stage that is a scan stage, we take the whole query's bytes read and add it. So if there are 2 scan stages - in that case, you will just return `2 * bytesreadbyquery`.
3. `getAllStages` does a DFS traversal every time - and given that it will be called in a loop, this might be an expensive operation and we don't really care about the ordering here.

In my opinion - we should do something similar to the existing logic that exists for `getTotalCpuTime`. That will help ensure that the logics are similar and it is easy to change them together if we ever decide to head that path.

Reply:
ok so this is newly implemented, the original proposal was a simple approach which is to obtain the scanned data size from `queryInfo::queryStats` if the final resolution is to do something similiar to the existing logic with `getTotalCpuTime`.
`DataSize scan = query.getQueryInfo().getQueryStats().getRawInputDataSize();`
@mbasmanova what's your suggestion on this one?

Reply:
`query.getQueryInfo().getQueryStats()` is a sort of expensive method since it aggregates all the stats and not just bytes read. We should keep this data collection as light as possible in my opinion.

Reply:
Agree with @mayankgarg1990 -- this can be made to be less expensive and consistent with how we calculate CPU time.

Reply:
thank you @mayankgarg1990 @tdcmeehan for the comment, will revise this code accordingly to align with general data collection logic

Reply:
Were these comments addressed?

Reply:
yes this code block has been removed as to make it simple and similar to how cpu limits enforcement is handled